fix test teardown race with ongoing test operation(client side fix)

temichus opened the pull request:
this commit is a client-side solution to fix 
fixes: https://github.com/scylladb/scylladb/issues/16472
fixes: https://github.com/scylladb/scylladb/issues/16651

In the case of having a runaway asyncio task which waits for a long time and then sends some requests.

server-side solution: https://github.com/scylladb/scylladb/pull/18236

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/8628/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/8628/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/8628/)
#### Build Details:
- Duration: 4 hr 58 min
- Builder: i-0a847770e37867c6b (m5ad.12xlarge)

kbr-scylla commented:
I think this should be part of https://github.com/scylladb/scylladb/pull/18236, because it fixes a problem that https://github.com/scylladb/scylladb/pull/18236 has (i.e. the runaway asyncio task problem).


Other than that I have two comments regarding commit message:
- forgot `:` after module: i.e. it should be `test/topology: ` not `test/topology `
- should use imperative mood: "function scope ManagerClient object " -> "make ManagerClient object function scope"

so please close this PR, add these two commits on top of https://github.com/scylladb/scylladb/pull/18236, and fix the commit messages.

kbr-scylla commented:
Also typo in second commit message, `Intriduce`

kbr-scylla commented on test/pylib/manager_client.py:
```diff
@@ -53,9 +53,12 @@ def __init__(self, sock_path: str, port: int, use_ssl: bool, auth_provider: Any|
 
     @property
     def client(self):
+        if self.test_finished_event.is_set():
```
What if some task first obtains `client`, then goes to sleep, then we call `after_test`, then the task resumes?

temichus replied:
yes. it is also possible. 
usually, test obtains ```manager``` and use ```client``` as ```manager.client``` and there is an actual workaround for this case

anyway obtaining ```client``` may affect only current current case because all ```clients``` will be closed  in ```manager``` fixture with scope function

as an additional workaround, we can close all clients in after_test->create new during after_test->close again after after_test. But I don't think it is worth it.

kbr-scylla replied:
Let's write a comment about this case, but explain why it's not important / not probable.

I hope we eventually migrate away from the client-rest-server architecture for topology tests, and all of this we'll be able to solve in a much cleaner way.

temichus commented:
> Also typo in second commit message, `Intriduce`
> 
> edit: there are more typos, please fix them all



> Also typo in second commit message, `Intriduce`
> 
> edit: there are more typos, please fix them all

fixed here https://github.com/scylladb/scylladb/pull/18236

temichus commented:
> I think this should be part of #18236, because it fixes a problem that #18236 has (i.e. the runaway asyncio task problem).
> 
> Other than that I have two comments regarding commit message:
> 
> * forgot `:` after module: i.e. it should be `test/topology: ` not `test/topology `
> * should use imperative mood: "function scope ManagerClient object " -> "make ManagerClient object function scope"
> 
> so please close this PR, add these two commits on top of #18236, and fix the commit messages.



> I think this should be part of #18236, because it fixes a problem that #18236 has (i.e. the runaway asyncio task problem).
> 
> Other than that I have two comments regarding commit message:
> 
> * forgot `:` after module: i.e. it should be `test/topology: ` not `test/topology `
> * should use imperative mood: "function scope ManagerClient object " -> "make ManagerClient object function scope"
> 
> so please close this PR, add these two commits on top of #18236, and fix the commit messages.


Fixed 
added commits to https://github.com/scylladb/scylladb/pull/18236 PR
